tfhe/src/integer/server_key/radix/scalar_mul.rs

```diff
@@ -1,11 +1,40 @@
-use crate::core_crypto::prelude::UnsignedInteger;
+use crate::core_crypto::prelude::{Numeric, UnsignedInteger};
 use crate::integer::block_decomposition::{BlockDecomposer, DecomposableInto};
 use crate::integer::ciphertext::RadixCiphertext;
 use crate::integer::server_key::CheckError;
 use crate::integer::server_key::CheckError::CarryFull;
-use crate::integer::ServerKey;
+use crate::integer::{ServerKey, U256};
 use std::collections::BTreeMap;
 
+pub trait ScalarMultiplier: Numeric {
+    fn is_power_of_two(self) -> bool;
+
+    fn ilog2(self) -> u32;
+}
```

Comment:
Not sure, as this is very specific to multiplication, it might be best to leave the definition closer to where its actually used
as its only going to be used here.

Ideally by the end of the next quarter we will have U256, I256, U512, I512 , etc and so we could go back and remove these traits and use The UnsignedInteger one

Reply:
ok it was just regarding the pub(super) thing